LanguageFeatures/Extension-types/exhaustiveness_cast_A01_t01.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The lifted space union for a pattern with matched value type M is
+/// ...
+/// Cast pattern:
+/// ...
+/// Let S be the lifted space union of the cast's subpattern in context C.
+/// i. If C is a subset (see below about type subsetting) of S then the result
+///   spaces is the lifted space union of M.
+/// ii. Otherwise, the result spaces is S plus the lifted space union of Null
+///   when C is a non-nullable type, and spaces is S when C is potentially
+///   nullable.
+///
+/// @description Check a lifted space of a cast pattern in case of not sealed
+/// type. Test switch statement
+/// @author sgrekhov22@gmail.com
+/// @issue 54460
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+extension type ObjectET1(Object _) {}
+extension type ObjectET2(Object _) implements Object {}
+
+extension type IntET1(int _) {}
+extension type IntET2(int _) implements int {}
+
+int test1_1(ObjectET1 obj) {
+  switch (obj) {
+    case int(isEven: true) as int:
+      return 1;
+    case int _:
+      return 2;
+  }
```

Comment:
This one is blocked on https://github.com/dart-lang/language/issues/3534: If I change `ObjectET1` to `Object` then the function has no error (that is, flow analysis then recognizes that it is not possible to reach the end of the function). This implies that (1) the implementation does not use the rule which is recommended in said issue, and (2) if that proposal is implemented then we won't have an error (so it does matter).

Reply:
Right, `ObjectET1` is not always-exhaustive, so we don't check the switch statement for exhaustiveness and we expect to be able to reach the end of the function, and that's an error. Good!